mediathread/assetmgr/views.py

```diff
@@ -1118,25 +1117,31 @@ def add_metadata(self, request, assets):
         return {'active_tags': tags, 'active_vocabulary': vocab}
 
     def apply_pagination(self, assets, notes, offset, limit):
-        # sort the object list via aggregation on the visible notes
-        assets = notes.values('asset')
-        assets = assets.annotate(last_modified=Max('modified'))
-        assets = assets.order_by('-last_modified')
-
-        # slice the list
+        """
+        Returns an (assets, notes) tuple, limited by limit, and offset
+        by offset.
+        """
         assets = assets[offset:offset + limit]
-        ids = [a['asset'] for a in assets]
+        ids = [a.pk for a in assets]
         return (assets, notes.filter(asset__id__in=ids))
 
     def get(self, request):
         if (self.record_owner):
-            assets = Asset.objects.by_course_and_user(request.course,
-                                                      self.record_owner)
+            assets = Asset.objects.by_course_and_user(
+                request.course,
+                self.record_owner)
         else:
             assets = Asset.objects.by_course(request.course)
 
         (assets, notes) = self.visible_assets_and_notes(request, assets)
 
+        # Order by title, by default.
+        ordering = request.GET.get('order_by', 'title')
+        if ordering == 'title' or ordering == 'author':
```

Comment:
I'm seeing sort inconsistencies when the sort order is `-author` or `-title`. Likely lowercase needs to be handled for those cases as well.

Reply:
oh yeah, of course